src/python/pants/backend/java/target_types.py

```diff
@@ -22,6 +22,7 @@
 from pants.engine.unions import UnionMembership, UnionRule
 from pants.jvm.target_types import (
     JunitTestSourceField,
+    JvmCompatibleJdkVersionField,
```

Comment:
Naming nit: this is more than a version, since it also includes "which" JDK is being used.